solutions/standard/main.tf

```diff
@@ -103,3 +103,62 @@ module "redis" {
   service_credential_names      = var.service_credential_names
   backup_crn                    = var.backup_crn
 }
+
+# create a service authorization between Secrets Manager and the target service (Databases for Redis)
+resource "ibm_iam_authorization_policy" "secrets_manager_key_manager" {
+  count                       = var.skip_rd_sm_auth_policy || var.existing_secrets_manager_instance_crn == null ? 0 : 1
```

Comment:
This is different to elasticsearch which has a local variable to control the count here and on the next resource.